### PR TITLE
feat(multilingual-control): create new dynamic multilingual widget 

### PR DESCRIPTION
**Pull request by EKrasovskiy**
…instead old

### CODE REVIEW

**Review by grankovsky — DISMISSED** [review]

File: packages/components/xm-multilingual-control/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export { XmMultilingualModule } from '@xm-ngx/components/xm-multilingual-control/xm-multilingual.module';
```
Comment (Kaskyi) [inline packages/components/xm-multilingual-control/index.ts:1]: `./` instead of `@xm-ngx/components/`

**Review by Kaskyi — APPROVED** [review]

**Review by Kaskyi — DISMISSED** [review]

File: packages/components/xm-multilingual-control/index.ts (hunk 1 above)
Comment (EKrasovskiy) [reply packages/components/xm-multilingual-control/index.ts]: fixed

File: packages/components/xm-multilingual-control/xm-multilingual-control.component.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                set(baseOptions, 'input.selector', '@xm-ngx/components/xm-angular-editor-control');
+                this._options = baseOptions;
+                return;
+            }
+            set(baseOptions, 'input.selector', '@xm-ngx/components/text-control');
+            this._options = baseOptions;
+            return;
+        }
+
+        this._options = baseOptions;
```
Comment (Sumragen) [inline packages/components/xm-multilingual-control/xm-multilingual-control.component.ts:78]: Could be simplified by defining mapper variable with enum as a key.

File: packages/components/xm-multilingual-control/xm-multilingual-control.component.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    public languages$: Observable<string[]>;
+    public selectedLng: FormControl = new FormControl();
+    public multilingualGroup: FormGroup = new FormGroup({});
+
+    public _onChange: (v: MultilingualControlData) => void = () => undefined;
+    public _onTouched: (v?: unknown) => void = () => undefined;
```
Comment (Sumragen) [inline packages/components/xm-multilingual-control/xm-multilingual-control.component.ts:86]: Syntax with underscore always means that class propery should be private.

File: packages/components/xm-multilingual-control/xm-multilingual-control.component.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        });
+    }
+
+    public ngOnDestroy(): void {
+        takeUntilOnDestroyDestroy(this);
+    }
+
+    private initFormGroup(): void {
+        this.languageService.languages$()
+            .pipe(take(1))
```
Comment (Sumragen) [inline packages/components/xm-multilingual-control/xm-multilingual-control.component.ts:117]: Probably we could use static property to achieve that behavior

Comment (Sumragen) [reply packages/components/xm-multilingual-control/xm-multilingual-control.component.ts:117]: Also we could use component config in order to determine necessary languages.

**Review by Sumragen — CHANGES_REQUESTED** [review]

File: packages/components/xm-multilingual-control/xm-multilingual-control.component.ts [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+    public registerOnChange(fn: (_: MultilingualControlData) => void): void {
+        this._onChange = fn;
+    }
+
+    public registerOnTouched(fn: any): void {
+        this._onTouched = fn;
+    }
+
+    public setDisabledState?(isDisabled: boolean): void {
```
Comment (Kaskyi) [inline packages/components/xm-multilingual-control/xm-multilingual-control.component.ts:151]: question mark at the end ? the method returns `void`

File: packages/components/xm-multilingual-control/xm-multilingual-control.component.ts [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+@Component({
+    selector: 'xm-multilingual-control',
+    templateUrl: './xm-multilingual-control.component.html',
+    styles: [`
+        mat-button-toggle-group {
+            margin-bottom: 10px;
+        }
+
+        mat-label {
```
Comment (Kaskyi) [inline packages/components/xm-multilingual-control/xm-multilingual-control.component.ts:50]: The global styles may create an unpredictable behaviour. Make sure to wrap them with a base class like `.xm-multilingual-control`

**Review by Kaskyi — CHANGES_REQUESTED** [review]

**Comment by EKrasovskiy** [comment]
https://github.com/xm-online/xm-webapp/pull/1334

**Pull request closed without merging**